modules/offlinemaps/src/main/java/com/sygic/maps/offlinemaps/loader/MapLoaderGlobal.kt

```diff
@@ -0,0 +1,271 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sygic.maps.offlinemaps.loader
+
+import androidx.lifecycle.MutableLiveData
+import com.sygic.maps.uikit.views.common.livedata.SingleLiveEvent
+import com.sygic.maps.uikit.views.common.utils.logError
+import com.sygic.maps.uikit.views.common.utils.logWarning
+import com.sygic.sdk.map.MapLoader
+import com.sygic.sdk.map.listeners.MapInstallProgressListener
+import com.sygic.sdk.map.listeners.MapResumedInstallDoneListener
+import kotlinx.coroutines.*
+import java.util.concurrent.Executors
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.EmptyCoroutineContext
+
+object MapLoaderGlobal {
+    private val mapLoaderDispatcher = Executors.newSingleThreadExecutor().asCoroutineDispatcher()
+    private val scope = CoroutineScope(mapLoaderDispatcher + SupervisorJob())
+
+    val continentsObservable = MutableLiveData<MutableMap<String, MutableList<String>>>()
+    val countriesObservable = MutableLiveData<MutableMap<String, CountryHolder>>()
+    val regionsObservable = MutableLiveData<MutableMap<String, RegionHolder>>()
+
+    val installedCountriesObservable = MutableLiveData<MutableMap<String, CountryHolder>>()
+    val installedRegionsObservable = MutableLiveData<MutableMap<String, RegionHolder>>()
+    val updatesAvailableObservable = MutableLiveData<List<String>>()
+    val detectedCountryObservable = SingleLiveEvent<CountryHolder>()
+
+    val notifyMapChangedObservable = SingleLiveEvent<Pair<String, MapLoader.MapStatus>>()
+    val mapInstallProgressObservable = MutableLiveData<Pair<String, Int>>()
+
+    val mapLoaderExceptionObservable = SingleLiveEvent<String>()
+
+    private val mapInstallProgressListener = MapInstallProgressListener { iso, downloadedBytes, totalSize ->
+        val progress = downloadedBytes / totalSize.toFloat()
+        val progressInt = (progress * 100).toInt()
+        updateMapsProgress(iso, progressInt)
+        mapInstallProgressObservable.value = iso to progressInt
+    }
+
+    private val mapResumedInstallDoneListener = MapResumedInstallDoneListener { iso, result ->
+        if (result != MapLoader.LoadResult.Success) {
+            logError("Failed resumed install done for $iso with result $result")
+        }
+        scope.launch {
+            val status = MapLoaderWrapper.getMapStatus(iso)
+            notifyMapChangedObservable.postValue(iso to status)
+        }
+    }
+
+    init {
+        scope.launch {
+            MapLoaderWrapper.getMapLoader().addMapProgressInstallListener(mapInstallProgressListener)
+            MapLoaderWrapper.getMapLoader().addMapResumedInstallDoneListener(mapResumedInstallDoneListener)
```

Comment:
listenery nikde neodoberame, ale asi to nevadi, kedze tento objekt zije cely cas, ze?

Reply:
Ano, tento objekt zije pocas celeho zivota aplikacie, takze si myslim ze by to nemalo vadit.